runtime/autoload/man.vim

```diff
@@ -328,18 +328,29 @@ function! man#complete(arg_lead, cmd_line, cursor_pos) abort
   return s:complete(sect, sect, name)
 endfunction
 
-function! s:get_paths(sect, name) abort
+function! s:get_paths(sect, name, do_fallback) abort
+  " callers must try-catch this, as some `man` implementations don't support `s:find_arg`
   try
     let mandirs = join(split(s:system(['man', s:find_arg]), ':\|\n'), ',')
+    return globpath(mandirs,'man?/'.a:name.'*.'.a:sect.'*', 0, 1)
   catch
-    call s:error(v:exception)
-    return
+    if !a:do_fallback
+      throw v:exception
+    endif
+
+    " fallback to a single path, with the page we're trying to find
+    let [l:sect, l:name, l:path] = s:verify_exists(a:sect, a:name)
+    return [l:path]
   endtry
-  return globpath(mandirs,'man?/'.a:name.'*.'.a:sect.'*', 0, 1)
 endfunction
 
 function! s:complete(sect, psect, name) abort
-  let pages = s:get_paths(a:sect, a:name)
+  try
+    let pages = s:get_paths(a:sect, a:name, v:false)
+  catch
+    call s:error(v:exception)
+    return
```

Comment:
why neuter the exception like this? Since we're showing an error anyway, might as well let the actual exception raise.  stacktrace is useful.

Reply:
Ah, I'd moved this bit of code from the exception handler in `s:get_paths()` - this snippet was present [before all the tag changes](https://github.com/neovim/neovim/blob/53b025887e28888f8dba78ff57afc001d1a6428b/runtime/autoload/man.vim#L367). I'm happy to let the exception propagate through, as the [introducing PR](https://github.com/neovim/neovim/pull/5734/files#diff-2451f3ce2ccfcf4300067af16309cd75R285) for this doesn't seem to have had a strong reason for adding it.

Reply:
iirc I added this as if man did throw an error, it was a little nicer to properly write the error out versus throwing an exception. The stack trace while helpful, isn't necessary as the error message will clearly indicate what happened since the codebase is small.